src/components/RadioGroup/RadioGroup.tsx

```diff
@@ -48,8 +49,11 @@ function RadioGroup(props: InternalRadioGroupProps) {
     throw new Error("RadioGroup options are invalid");
   }
 
-  const [labelId] = useState(() => `radio-group-label-${nanoid()}`);
-  const [auxId] = useState(() => `radio-group-aux-${nanoid()}`);
+  const labelId = useMemo(() => `radio-group-label-${nanoid()}`, []);
+  const auxId = useMemo(() => `radio-group-aux-${nanoid()}`, []);
```

Comment:
Curious why you changed useState to useMemo?

Reply:
Because this is not really state, all we want is a static Id which is persistent throughout the component life. So the correct use would be `useMemo` and not `useState` we are not even doing setState and so I updated it